bin/wserver/wserver.ml

```diff
@@ -1,100 +1,188 @@
 (* Copyright (c) 1998-2007 INRIA *)
 
-let connection_closed = ref false
-
-let eprintf = Printf.eprintf
-
-let sock_in = ref "wserver.sin"
-let sock_out = ref "wserver.sou"
+#ifdef WINDOWS
+(* temporary files for process exchange *)
+let in_fname = ref ("~" ^ (Filename.remove_extension (Filename.basename Sys.executable_name)) ^ "_post_rq.txt")
+let out_fname = ref ("~" ^ (Filename.remove_extension (Filename.basename Sys.executable_name)) ^ "_response.txt")
+#endif 
+  
 let stop_server = ref "STOP_SERVER"
 let noproc = ref false
 let cgi = ref false
 
 let wserver_sock = ref Unix.stdout
-let wsocket () = !wserver_sock
-
-let wserver_oc = ref stdout
-let woc () = !wserver_oc
-
-let wflush () = flush !wserver_oc
-
-let skip_possible_remaining_chars fd =
-  let b = Bytes.create 3 in
-  try
-    let rec loop () =
-      match Unix.select [fd] [] [] 5.0 with
-      | [_], [], [] ->
-        let len = Unix.read fd b 0 (Bytes.length b) in
-        if len = Bytes.length b then loop ()
-      | _ -> ()
-    in
-    loop ()
-  with Unix.Unix_error (Unix.ECONNRESET, _, _) -> ()
-
-let close_connection () =
```

Comment:
What is wrong with this, and why is `shutdown_noerr` better?

Reply:
close_connection did several things 
skip_possible_remaining_chars was a useless function that only read empty stream   (caused unnecessary waiting time) . only a shutdown / close sequence is needed